prime-router/src/main/kotlin/credentials/CredentialService.kt

```diff
@@ -0,0 +1,24 @@
+package gov.cdc.prime.router.credentials
+
+import org.apache.logging.log4j.kotlin.Logging
+
+abstract class CredentialService : Logging {
+    internal abstract fun fetchCredential(connectionId: String): Credential?
+    internal abstract fun saveCredential(connectionId: String, credential: Credential)
+
+    fun fetchCredential(connectionId: String, callerId: String, reason: CredentialRequestReason): Credential? {
+        logger.info { "CREDENTIAL REQUEST: $callerId requested connectionId($connectionId) credential for $reason" }
+        return fetchCredential(connectionId)
+    }
+
+    fun saveCredential(connectionId: String, credential: Credential, callerId: String) {
+        logger.info { "CREDENTIAL UPDATE: $callerId updated connectionId($connectionId) credential" }
+        return saveCredential(connectionId, credential)
+    }
```

Comment:
Abstract class responsible for logging the credential request.

Reply:
I am anticipating in #304 `connectionId` will be a UUID that gets set in the `organizations.yml`, for example:

```
transports:
          - type: SFTP
            host: hssftp.azdhs.gov
            port: 22
            filePath: ./TEST/IN
            connectionId: 4078f2fb-27c9-43c4-80af-cb679b0abe31
```